### DISCUSSION TITLE
Lesson 2 | Attempting to deploy my very first Contract

### DISCUSSION THREAD

**Discussion by jhuntertx3859** [Cyfrin/foundry-full-course-cu#1512] [General]
Having difficulty changing from the REMIX VM (Shanghai) environment to Injected Provider - Metamask environment. When I select Injected - Metamask, switches back to REMIX M (Shanghai). Will not allow me to change.

**Comment by EngrPips** [comment]
Hello @jhuntertx3859 , I do have similar problem sometimes as well and i am not really sure why that happens but the solution to that problem is you should instead choose the `wallet connect` option as shown below 

<img width="1440" alt="Screenshot 2024-03-22 at 12 45 55 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/cb525fa3-8f6e-4855-81ac-f2f0d5fc38d7">

If you click that option it should show the below 

<img width="1440" alt="Screenshot 2024-03-22 at 12 46 09 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/5db20edb-595e-4e3b-ac48-3ac55b3cf8ae">

Then click on the `connect wallet`button and it should pop up a list of wallet you can connect to  as shown below 

<img width="1440" alt="Screenshot 2024-03-22 at 12 46 18 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/3448dc30-78ec-4252-91f1-b75d41f98aef">

Then select `metamask` and it should pop up your `metamask` and ask you to connect. if you connect successfully then it should reflect on remix as shown below 

<img width="1440" alt="Screenshot 2024-03-22 at 12 46 36 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/2dd45c4c-737b-44fb-be35-03fa64b7b8b5">

Let me know if that works for you.
